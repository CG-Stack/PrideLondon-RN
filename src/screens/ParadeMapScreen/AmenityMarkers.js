// @flow
import React from "react";
import { Marker } from "react-native-maps";
import type { Amenity } from "../../data/amenity";
import amenityIconFirstAid from "../../../assets/images/amenityIconFirstAid.png";
import amenityIconToilet from "../../../assets/images/amenityIconToilet.png";

type Props = {
  amenities: Amenity[]
};

const iconMap = {
  Toilet: amenityIconToilet,
  "First Aid": amenityIconFirstAid
};

const AmenityMarkers = ({ amenities }: Props) => {
  if (amenities.length === 0) {
    return null;
  }

  return amenities.map(amenity => (
    <Marker
      pointerEvents="none"
      coordinate={{
        longitude: amenity.fields.location.lon,
        latitude: amenity.fields.location.lat
      }}
      key={amenity.id}
      image={iconMap[amenity.fields.type]}
    />
  ));
};

export default AmenityMarkers;
