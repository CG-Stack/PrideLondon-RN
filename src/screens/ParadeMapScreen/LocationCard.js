// @flow
import React from "react";
import { Image, StyleSheet, View } from "react-native";
import ContentPadding from "../../components/ContentPadding";
import Text, { scaleWithFont } from "../../components/Text";
import locationCardDefault from "../../../assets/images/locationCardDefault.png";
import { blackTwentyColor, whiteColor } from "../../constants/colors";
import type { ImageRef } from "../../data/image-ref";

type Props = {
  name?: string,
  text?: string,
  image?: ImageRef | string
};

const LocationCard = ({ name, text, image }: Props) => (
  <ContentPadding
    padding={{
      small: { horizontal: 8, vertical: 8 },
      medium: { horizontal: 16, vertical: 16 },
      large: { horizontal: 20, vertical: 20 }
    }}
  >
    <View style={styles.container}>
      <View style={styles.eventCard}>
        <View style={styles.eventCardInnerWrapper}>
          <Image
            style={styles.imageContainer}
            source={image}
            resizeMode="cover"
          />
          <View style={styles.eventCardDetails}>
            <Text
              numberOfLines={1}
              type="h3"
              color="lightNavyBlueColor"
              style={styles.eventName}
            >
              {name}
            </Text>
            <Text numberOfLines={3} type="small" color="lightNavyBlueColor">
              {text}
            </Text>
          </View>
        </View>
      </View>
    </View>
  </ContentPadding>
);

LocationCard.defaultProps = {
  name: "Come back soon",
  text: "We are currently adding the events to the map. Check back later!",
  image: locationCardDefault
};

const styles = StyleSheet.create({
  container: {
    borderRadius: 5,
    // The below properties are required for ioS shadow
    shadowColor: blackTwentyColor,
    shadowOffset: { width: 0, height: 1 },
    shadowOpacity: 1,
    shadowRadius: 3,
    // The below properties are required for android shadow
    borderWidth: 0,
    elevation: 3
  },
  eventCard: {
    height: scaleWithFont("h3", 108),
    flexDirection: "row",
    overflow: "hidden",
    borderRadius: 5
  },
  eventCardInnerWrapper: {
    flex: 1,
    flexDirection: "row"
  },
  imageContainer: {
    width: 114
  },
  eventCardDetails: {
    flex: 1,
    paddingLeft: 8,
    paddingTop: 12,
    backgroundColor: whiteColor
  },
  eventName: {
    paddingTop: 4
  }
});

export default LocationCard;
