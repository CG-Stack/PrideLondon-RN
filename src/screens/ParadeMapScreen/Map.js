// @flow
import React, { Component } from "react";
import { StyleSheet, View, Image } from "react-native";
import MapView, { Polyline, Marker } from "react-native-maps";
import type { Event, SavedEvents } from "../../data/event";
import paradeCoordinates from "../../constants/parade-coordinates";
import {
  warmPinkColor,
  lightNavyBlueColor,
  transparent
} from "../../constants/colors";
import Text from "../../components/Text";
import EventCard from "../../components/EventCard";
import ContentPadding from "../../components/ContentPadding";
import stageIconActive from "../../../assets/images/stageIconActive.png";
import stageIconInactive from "../../../assets/images/stageIconInactive.png";

type Props = {
  stages: Array<Event>,
  savedEvents: SavedEvents,
  addSavedEvent: string => void,
  removeSavedEvent: string => void,
  onPress: (id: string) => void
};

type State = {
  activeMarker: ?string,
  tileDetails: ?Event
};

class Map extends Component<Props, State> {
  state = {
    activeMarker: null,
    tileDetails: null
  };

  handleMarkerPress(stage: Event) {
    this.setState({ tileDetails: stage, activeMarker: stage.id });
  }

  handleMapPress = () => {
    this.setState({ tileDetails: null, activeMarker: null });
  };

  renderStageMarker = (stage: Event) => (
    <Marker
      coordinate={{
        longitude: stage.fields.location.lon,
        latitude: stage.fields.location.lat
      }}
      key={stage.id}
      onPress={() => this.handleMarkerPress(stage)}
      stopPropagation={true}
    >
      <Image
        source={
          this.state.activeMarker === stage.id
            ? stageIconActive
            : stageIconInactive
        }
      />
    </Marker>
  );

  render() {
    const {
      savedEvents,
      addSavedEvent,
      removeSavedEvent,
      onPress
    } = this.props;
    return (
      <View style={styles.mapWrapper}>
        <MapView
          style={StyleSheet.absoluteFill}
          initialRegion={{
            latitude: 51.5085,
            longitude: -0.134192,
            latitudeDelta: 0.02,
            longitudeDelta: 0.02
          }}
          showsPointsOfInterest={false}
          showsScale={false}
          showsBuildings={false}
          showsTraffic={false}
          showsIndoors={false}
          onPress={this.handleMapPress}
        >
          <Polyline
            coordinates={paradeCoordinates}
            strokeWidth={5}
            strokeColor={warmPinkColor}
            lineJoin="bevel"
          />
          <Marker
            coordinate={{ longitude: -0.14223, latitude: 51.51616 }}
            centerOffset={{ x: 0, y: -15 }}
            anchor={{ x: 1, y: 1 }}
            stopPropagation={true}
          >
            <View style={styles.markerView}>
              <View style={styles.markerTextWrapper}>
                <Text type="h4" color="whiteColor">
                  Start
                </Text>
              </View>
              <View style={styles.triangle} />
            </View>
          </Marker>
          <Marker
            coordinate={{ longitude: -0.1265, latitude: 51.50499 }}
            centerOffset={{ x: 0, y: -15 }}
            anchor={{ x: 1, y: 1 }}
            stopPropagation={true}
          >
            <View style={styles.markerView}>
              <View style={styles.markerTextWrapper}>
                <Text type="h4" color="whiteColor">
                  Finish
                </Text>
              </View>
              <View style={styles.triangle} />
            </View>
          </Marker>
          {this.props.stages.length > 0 &&
            this.props.stages.map(stage => this.renderStageMarker(stage))}
        </MapView>
        {this.state.tileDetails && (
          <View style={styles.tileContainer}>
            <ContentPadding
              padding={{
                small: { horizontal: 8, vertical: 8 },
                medium: { horizontal: 16, vertical: 16 },
                large: { horizontal: 20, vertical: 20 }
              }}
            >
              <EventCard
                id={this.state.tileDetails.id}
                name={this.state.tileDetails.fields.name}
                locationName={this.state.tileDetails.fields.locationName}
                eventPriceLow={this.state.tileDetails.fields.eventPriceLow}
                eventPriceHigh={this.state.tileDetails.fields.eventPriceHigh}
                startTime={this.state.tileDetails.fields.startTime}
                endTime={this.state.tileDetails.fields.endTime}
                imageReference={this.state.tileDetails.fields.eventsListPicture}
                isSaved={savedEvents.has(this.state.tileDetails.id)}
                addSavedEvent={addSavedEvent}
                removeSavedEvent={removeSavedEvent}
                onPress={onPress}
              />
            </ContentPadding>
          </View>
        )}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  mapWrapper: {
    ...StyleSheet.absoluteFillObject,
    justifyContent: "flex-end",
    alignItems: "center"
  },
  markerView: {
    alignItems: "center"
  },
  markerTextWrapper: {
    paddingHorizontal: 8,
    paddingVertical: 2,
    backgroundColor: lightNavyBlueColor
  },
  tileContainer: {
    width: "100%"
  },
  triangle: {
    width: 0,
    height: 0,
    backgroundColor: transparent,
    borderStyle: "solid",
    borderLeftWidth: 5,
    borderRightWidth: 5,
    borderBottomWidth: 10,
    borderLeftColor: transparent,
    borderRightColor: transparent,
    borderBottomColor: lightNavyBlueColor,
    transform: [{ rotate: "180deg" }]
  }
});

export default Map;
