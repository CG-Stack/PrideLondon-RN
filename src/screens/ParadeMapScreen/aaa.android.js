// @flow
import React, { Component } from "react";
import type { ElementRef } from "react";
import { View, StyleSheet } from "react-native";
import MapView, { Polyline, Marker } from "react-native-maps";
import Text from "../../components/Text";
import { velvetColor } from "../../constants/colors";
import type {
  Coordinates,
  Region,
  Terminals
} from "../../constants/parade-coordinates";

type Props = {
  route: Array<Coordinates>,
  paradeRegion: Region,
  terminals: Array<Terminals>,
  permission: boolean
};

type State = {
  margin: boolean
};

class Map extends Component<Props, State> {
  constructor() {
    super();

    this.state = {
      margin: false
    };
  }

  componentWillMount() {
    setTimeout(() => this.setState({ margin: !this.state.margin }), 500);
  }

  componentDidUpdate(prevProps: Props) {
    if (prevProps.permission !== this.props.permission) {
      setTimeout(() => this.setState({ margin: !this.state.margin }), 500);
    }
  }

  focus(region: Region) {
    this.mapView.animateToRegion(region, 0);
  }

  // $FlowFixMe
  mapView: ElementRef;

  render() {
    return (
      <View style={styles.absoluteFill}>
        <MapView
          style={[
            styles.absoluteFill,
            this.state.margin ? styles.marginTrue : styles.marginFalse
          ]}
          initialRegion={this.props.paradeRegion}
          showsUserLocation={this.props.permission}
          showsUserLocationButton={this.props.permission}
          ref={component => {
            this.mapView = component;
          }}
        >
          <Polyline
            coordinates={this.props.route}
            strokeWidth={5}
            strokeColor={velvetColor}
            lineJoin="bevel"
          />
          {this.props.terminals.map(terminal => (
            <Marker coordinate={terminal.coordinates} key={terminal.key}>
              <View style={terminal.style}>
                <Text type={terminal.text.type} color={terminal.text.color}>
                  {terminal.text.text}
                </Text>
              </View>
            </Marker>
          ))}
        </MapView>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  absoluteFill: {
    ...StyleSheet.absoluteFillObject
  },
  marginTrue: {
    margin: 1
  },
  marginFalse: {
    margin: 0
  }
});

export default Map;
