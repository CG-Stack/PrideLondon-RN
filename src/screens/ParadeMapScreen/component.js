// @flow
import React from "react";
import { StyleSheet, View, Text } from "react-native";
import MapView, { Polyline, Marker } from "react-native-maps";

import paradeCoordinates from "../../constants/parade-coordinates";
import { velvetColor, whiteColor } from "../../constants/colors";

const ParadeMap = () => (
  <MapView
    style={StyleSheet.absoluteFill}
    scrollEnabled={false}
    region={{
      latitude: 51.51004,
      longitude: -0.134192,
      latitudeDelta: 0.02,
      longitudeDelta: 0.0112
    }}
  >
    <Polyline
      coordinates={paradeCoordinates}
      strokeWidth={5}
      strokeColor={velvetColor}
      lineJoin="bevel"
    />
    <Marker coordinate={{ longitude: -0.14223, latitude: 51.51616 }}>
      <View style={styles.markerView}>
        <Text style={styles.markerText}>A</Text>
      </View>
    </Marker>
    <Marker coordinate={{ longitude: -0.1265, latitude: 51.50499 }}>
      <View style={styles.markerView}>
        <Text style={styles.markerText}>B</Text>
      </View>
    </Marker>
  </MapView>
);

const styles = StyleSheet.create({
  markerView: {
    height: 15,
    width: 15,
    backgroundColor: velvetColor,
    borderRadius: 7,
    justifyContent: "center",
    alignItems: "center"
  },
  markerText: {
    color: whiteColor,
    fontSize: 12
  }
});

export default ParadeMap;
