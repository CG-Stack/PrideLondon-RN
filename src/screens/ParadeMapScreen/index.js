// @flow
import { connect } from "react-redux";
import type { Connector } from "react-redux";
import type { State } from "../../reducers";
import withIsFocused from "../../components/WithIsFocused";
import Component from "./component";
import type { Event } from "../../data/event";
import { selectStages } from "../../selectors";
import { networkInterfaces } from "os";

type OwnProps = {
  isFocused: boolean
};

type StateProps = {
  stages: Event[],
  isFocused: boolean
};

let cache: StateProps;

// Note we must add a return type here for react-redux connect to work
// with flow correctly. If not provided is silently fails if types do
// not line up. See https://github.com/facebook/flow/issues/5343
const mapStateToProps = (state: State, { isFocused }: OwnProps): StateProps => {
  if (!cache || isFocused) {
    cache = {
      stages: selectStages(state),
      isFocused
    };
  }
  return cache;
};

const connector: Connector<OwnProps, StateProps> = connect(mapStateToProps);

export const Container = connector(Component);

export default withIsFocused(Container);
