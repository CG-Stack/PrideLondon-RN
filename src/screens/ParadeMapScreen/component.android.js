// @flow
import React, { Component } from "react";
import { StyleSheet, View } from "react-native";
import MapView, { Polyline, Marker } from "react-native-maps";
import Permissions from "react-native-permissions";
import paradeCoordinates from "../../constants/parade-coordinates";
import { velvetColor, whiteColor } from "../../constants/colors";
import Text, { scaleWithFont } from "../../components/Text";
import Touchable from "../../components/Touchable";
import LocationCard from "./LocationCard";

class ParadeMapScreen extends Component {
  constructor() {
    super();
    this.state = {
      locationPermission: false
    };
  }

  componentDidMount() {
    this.checkPermission();
  }

  checkPermission = () => {
    Permissions.check("location").then(response => {
      if (response === "authorized")
        this.setState({ locationPermission: true });
      if (response === "denied" || response === "undetermined")
        this.requestPermission();
    });
  };

  requestPermission = () => {
    Permissions.request("location").then(response => {
      this.setState({ locationPermission: true });
    });
  };

  render() {
    return (
      <View style={styles.container} testID="parade-map-screen">
        <MapView
          // second style object to force re-render in order to display location button
          style={[
            StyleSheet.absoluteFillObject,
            this.state.locationPermission && styles.mapView
          ]}
          region={{
            latitude: 51.5085,
            longitude: -0.134192,
            latitudeDelta: 0.02,
            longitudeDelta: 0.000000041
          }}
          showsUserLocation={this.state.locationPermission}
          showsMyLocationButton={this.state.locationPermission}
        >
          <Polyline
            coordinates={paradeCoordinates}
            strokeWidth={5}
            strokeColor={velvetColor}
            lineJoin="bevel"
          />

          <Marker coordinate={{ longitude: -0.14223, latitude: 51.51616 }}>
            <View style={styles.markerView}>
              <Text type="xSmall" color="whiteColor">
                A
              </Text>
            </View>
          </Marker>
          <Marker coordinate={{ longitude: -0.1265, latitude: 51.50499 }}>
            <View style={styles.markerView}>
              <Text type="xSmall" color="whiteColor">
                B
              </Text>
            </View>
          </Marker>
        </MapView>
        <LocationCard />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    ...StyleSheet.absoluteFillObject,
    flexDirection: "column",
    justifyContent: "flex-end"
  },
  mapView: {
    marginTop: -1
  },
  markerView: {
    height: Math.max(15, scaleWithFont("xSmall", 18)),
    width: Math.max(15, scaleWithFont("xSmall", 18)),
    backgroundColor: velvetColor,
    borderRadius: Math.max(9, scaleWithFont("xSmall", 9)),
    justifyContent: "center",
    alignItems: "center"
  }
});

export default ParadeMapScreen;
