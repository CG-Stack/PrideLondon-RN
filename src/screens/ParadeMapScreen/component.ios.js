// @flow
import React, { Component } from "react";
import { StyleSheet, View, Image } from "react-native";
import MapView, { Polyline, Marker } from "react-native-maps";
import Permissions from "react-native-permissions";
import type {
  NavigationEventSubscription,
  NavigationScreenProp,
  NavigationState
} from "react-navigation";
import { route, region } from "../../constants/parade-coordinates";
import { velvetColor, whiteColor } from "../../constants/colors";
import Text, { scaleWithFont } from "../../components/Text";
import Touchable from "../../components/Touchable";
import LocationCard from "./LocationCard";

import locationButtonInactive from "../../../assets/images/location-inactive.png";
import locationButtonActive from "../../../assets/images/location-active.png";

type Props = {
  navigation: NavigationScreenProp<NavigationState>
};

class ParadeMapScreen extends Component<Props> {
  constructor(props) {
    super(props);
    this.state = {
      locationPermission: false,
      atUserLocation: false
    };
  }

  componentDidMount() {
    this.checkPermission();
  }

  checkPermission = () => {
    Permissions.check("location").then(response => {
      if (response === "authorized")
        this.setState({ locationPermission: true });
      if (response === "denied" || response === "undetermined")
        this.requestPermission();
    });
  };

  requestPermission = () => {
    Permissions.request("location").then(response => {
      this.setState({ locationPermission: true });
    });
  };

  moveToCurrentLocation = () => {
    navigator.geolocation.getCurrentPosition(({ coords }) => {
      const { latitude, longitude } = coords;
      this._map.animateToCoordinate(
        {
          latitude,
          longitude
        },
        500
      );
    });
  };

  onRegionChange = position => {
    const { latitude: currentLat, longitude: currentLong } = position;
    navigator.geolocation.getCurrentPosition(({ coords }) => {
      const { latitude: userLat, longitude: userLong } = coords;
      if (
        userLat.toFixed(5) === currentLat.toFixed(5) &&
        userLong.toFixed(5) === currentLong.toFixed(5)
      ) {
        this.setState({ atUserLocation: true });
      }
    });
  };

  render() {
    return (
      <View style={styles.container} testID="parade-map-screen">
        <MapView
          style={StyleSheet.absoluteFill}
          initialRegion={region}
          showsUserLocation={this.state.locationPermission}
          ref={component => {
            this._map = component;
          }}
          onRegionChange={this.onRegionChange}
          key={this.props.navigation.isFocused()}
        >
          <Polyline
            coordinates={route}
            strokeWidth={5}
            strokeColor={velvetColor}
            lineJoin="bevel"
          />

          <Marker coordinate={{ longitude: -0.14223, latitude: 51.51616 }}>
            <View style={styles.markerView}>
              <Text type="xSmall" color="whiteColor">
                A
              </Text>
            </View>
          </Marker>
          <Marker coordinate={{ longitude: -0.1265, latitude: 51.50499 }}>
            <View style={styles.markerView}>
              <Text type="xSmall" color="whiteColor">
                B
              </Text>
            </View>
          </Marker>
        </MapView>
        {this.state.locationPermission && (
          <Touchable
            onPress={this.moveToCurrentLocation}
            style={{ alignSelf: "flex-end" }}
          >
            <Image
              source={
                this.state.atUserLocation
                  ? locationButtonActive
                  : locationButtonInactive
              }
            />
          </Touchable>
        )}
        {/* <LocationCard /> */}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    ...StyleSheet.absoluteFillObject,
    flexDirection: "column",
    justifyContent: "flex-end"
  },
  markerView: {
    height: Math.max(15, scaleWithFont("xSmall", 18)),
    width: Math.max(15, scaleWithFont("xSmall", 18)),
    backgroundColor: velvetColor,
    borderRadius: Math.max(9, scaleWithFont("xSmall", 9)),
    justifyContent: "center",
    alignItems: "center"
  }
});

export default ParadeMapScreen;
