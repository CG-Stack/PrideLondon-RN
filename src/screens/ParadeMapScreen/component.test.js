// @flow
import React from "react";
import { shallow } from "enzyme";
import type { NavigationScreenProp, NavigationState } from "react-navigation";
import { generateEvent, sampleOne } from "../../data/__test-data";
import { EVENT_DETAILS } from "../../constants/routes";
import ParadeMapScreen from "./component";
import Map from "./Map";

const navigation: NavigationScreenProp<NavigationState> = ({
  navigate: () => {}
}: any);

const stages = sampleOne(generateEvent, { seed: 5728 });

it("renders correctly", () => {
  const output = shallow(
    <ParadeMapScreen
      isFocused
      navigation={navigation}
      stages={[stages]}
      addSavedEvent={() => {}}
      removeSavedEvent={() => {}}
      onPress={() => {}}
      savedEvents={new Set()}
    />
  );
  expect(output).toMatchSnapshot();
});

it("does not render map when not focused", () => {
  const output = shallow(
    <ParadeMapScreen
      isFocused={false}
      navigation={navigation}
      stages={[stages]}
      addSavedEvent={() => {}}
      removeSavedEvent={() => {}}
      onPress={() => {}}
      savedEvents={new Set()}
    />
  );
  expect(output).toMatchSnapshot();
});

it("opens an event", () => {
  const navigationSpy = jest.fn();
  const nav: NavigationScreenProp<NavigationState> = ({
    navigate: navigationSpy
  }: any);
  const output = shallow(
    <ParadeMapScreen
      isFocused
      navigation={nav}
      stages={[stages]}
      addSavedEvent={() => {}}
      removeSavedEvent={() => {}}
      onPress={() => {}}
      savedEvents={new Set()}
    />
  );

  output
    .find(Map)
    .props()
    .onPress(1);

  expect(navigationSpy).toBeCalledWith(EVENT_DETAILS, { eventId: 1 });
  jest.clearAllMocks();
});
