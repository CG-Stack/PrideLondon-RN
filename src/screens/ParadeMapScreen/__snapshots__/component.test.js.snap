// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not render map when not focused 1`] = `
<Component
  style={
    Object {
      "bottom": 0,
      "flexDirection": "column",
      "justifyContent": "flex-end",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
  testID="parade-map-screen"
>
  <LocationCard
    image={1}
    name="Come back soon"
    text="We are currently adding the events to the map. Check back later!"
  />
</Component>
`;

exports[`renders correctly 1`] = `
<Component
  style={
    Object {
      "bottom": 0,
      "flexDirection": "column",
      "justifyContent": "flex-end",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
  testID="parade-map-screen"
>
  <Map />
  <LocationCard
    image={1}
    name="Come back soon"
    text="We are currently adding the events to the map. Check back later!"
  />
</Component>
`;
